packages/flutter/test/widgets/page_view_test.dart

```diff
@@ -565,6 +565,52 @@ void main() {
     expect(tester.getTopLeft(find.text('Hawaii')), const Offset(-100.0, 0.0));
   });
 
+  testWidgets('All visible pages are able to receive touch events',
+    (WidgetTester tester) async {
+      final PageController controller = PageController(viewportFraction: 1/4, initialPage: 0);
+      int tappedIndex;
+      Iterable<int> visiblePages;
+
+      Widget build() {
+        return Directionality(
+          textDirection: TextDirection.ltr,
+          child: PageView.builder(
+            controller: controller,
+            itemCount: 20,
+            itemBuilder: (BuildContext context, int index) {
+              return GestureDetector(
+                onTap: () => tappedIndex = index,
+                child: SizedBox.expand(child: Text(index.toString())),
+              );
+            },
+          ),
+        );
+      }
+
+      visiblePages = const <int> [0, 1, 2];
+      await tester.pumpWidget(build());
+      // The first 3 items should be visible and tappable.
+
+      for (int index in visiblePages) {
+        expect(find.text(index.toString()), findsOneWidget);
+        // The center of 3 has an x-coordinate of 800, so we have to manually
```

Comment:
What do you mean by this?

Reply:
Does sound ambiguous after re-reading the comment. Updated.